actors/builtin/shared.go

```diff
@@ -61,20 +54,28 @@ func RT_GetMinerAccountsAssert(rt runtime.Runtime, minerAddr addr.Address) (owne
 	return
 }
 
-func RT_MinerEntry_ValidateCaller_DetermineFundsLocation(rt runtime.Runtime, entryAddr addr.Address, entrySpec MinerEntrySpec) addr.Address {
-	if RT_Address_Is_StorageMiner(rt, entryAddr) {
+func MarketAddress(rt runtime.Runtime, addr addr.Address) addr.Address {
+	if IsStorageMiner(rt, addr) {
 		// Storage miner actor entry; implied funds recipient is the associated owner address.
-		ownerAddr, workerAddr := RT_GetMinerAccountsAssert(rt, entryAddr)
+		ownerAddr, workerAddr := GetMinerCtl(rt, addr)
 		rt.ValidateImmediateCallerIs(ownerAddr, workerAddr)
-		return ownerAddr
-	} else {
-		if entrySpec == MinerEntrySpec_MinerOnly {
-			rt.Abort(exitcode.ErrPlaceholder, "Only miner entries valid in current context")
-		}
-		// Ordinary account-style actor entry; funds recipient is just the entry address itself.
-		rt.ValidateImmediateCallerType(CallerTypesSignable...)
-		return entryAddr
+		return ownerAddr // TODO: That should probably be the worker address
 	}
+
+	// Ordinary account-style actor entry; funds recipient is just the entry address itself.
+	rt.ValidateImmediateCallerType(CallerTypesSignable...)
+	return addr
+}
+
+func PledgeAddress(rt runtime.Runtime, addr addr.Address) addr.Address {
+	if !IsStorageMiner(rt, addr) {
+		rt.Abort(exitcode.ErrPlaceholder, "Only miner entries valid in current context")
+	}
+
+	// Storage miner actor entry; implied funds recipient is the associated owner address.
+	ownerAddr, workerAddr := GetMinerCtl(rt, addr)
+	rt.ValidateImmediateCallerIs(ownerAddr, workerAddr)
+	return ownerAddr // TODO: this probably just be addr, otherwise changing keys is really annoying
```

Comment:
Thanks for cleaning this up, it was horrible code.

The owner address in each case has been discussed at length and is probably the right choice for now. We can discuss though - please remove the TODOs and file an issue if desired.